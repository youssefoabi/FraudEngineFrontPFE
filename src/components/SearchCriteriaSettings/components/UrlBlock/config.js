// eslint-disable-next-line import/prefer-default-export
export const config = [
  {
    columns: 8,
    id: 'name',
    type: 'input',
    label: 'Libelle du Radio-Bouton',
  },
  {
    columns: 4,
    id: 'isEnable',
    type: 'switch',
    defaultValue: false,
    label: 'Gamme activ√©e',
  },
];
