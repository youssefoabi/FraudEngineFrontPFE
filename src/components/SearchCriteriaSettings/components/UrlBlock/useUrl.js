import { isEmpty, isNil, map, pathOr, prop } from 'ramda';
import { useCallback, useEffect, useMemo, useState } from 'react';

import clientApi from '../../../../client-api';

export default function useUrl({ engine, handleChange }) {
  const [scope, setScope] = useState([]);

  const { scopes, currentScope } = engine || {};

  const scopeId = pathOr({}, ['scopeId'], currentScope);
  const scopeLabel = pathOr('', [scopeId, 'name'], scopes);
  const { parameters: allParameters, urlTemplate: scopeUrl } = scope || {};

  useEffect(() => {
    if (isNil(scopeId)) return;

    clientApi()
      .getScope(scopeId)
      .then(prop('data'))
      .then(scopeById => {
        setScope(scopeById);
      });
  }, [scopeId]);

  const mapParameter = parameter => {
    if (isNil(parameter) || isEmpty(parameter)) return null;

    const { key } = parameter;
    return key;
  };

  const mapParameters = useCallback(parameters => {
    if (isNil(parameters) || isEmpty(parameters)) return null;

    return map(mapParameter, parameters);
  }, []);

  const mappedParameters = useMemo(() => mapParameters(allParameters), [allParameters, mapParameters]);

  const handleScopeChange = ({ id, value }) => {
    handleChange({
      id: 'scopes',
      value: { ...(scopes || {}), [scopeId]: { ...scopes[scopeId], [id]: value } },
    });
  };

  return {
    scopeUrl,
    scopeId,
    scopeLabel,
    allParameters: mappedParameters,
    handleScopeChange,
  };
}
