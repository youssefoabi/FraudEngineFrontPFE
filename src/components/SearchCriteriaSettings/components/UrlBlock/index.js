import { DataEdit, GridItem } from '@cdiscount/backoffice-ui';
import { FormLabel, Grid } from '@material-ui/core';
import { func, shape } from 'prop-types';
import { isNilOrEmpty } from 'ramda-adjunct';
import { pathOr } from 'ramda';
import React from 'react';

import { config } from './config';
import ParameterChip from './components/ParameterChip';
import ScopeUrlField from './components/ScopeUrlField';
import useUrl from './useUrl';

const UrlBlock = ({ engine, handleChange }) => {
  const { scopeId, scopeUrl, scopeLabel, allParameters, handleScopeChange } = useUrl({
    engine,
    handleChange,
  });
  const scope = pathOr(null, ['scopes', scopeId], engine);

  const renderParametersList = () => {
    if (isNilOrEmpty(allParameters)) return null;

    return allParameters.map(currentParameter => (
      <ParameterChip engine={engine} scopeId={scopeId} parameter={currentParameter} />
    ));
  };

  return (
    <GridItem title={`Gamme ${scopeLabel || ''}`}>
      <Grid container spacing={2}>
        <Grid item columns={6} xs={6} md={6} lg={6}>
          <FormLabel>Url Pattern</FormLabel>
          <ScopeUrlField scopeUrl={scopeUrl} />
        </Grid>
        <Grid item columns={6} xs={6} md={6} lg={6}>
          <DataEdit fields={config} values={scope} onChange={handleScopeChange} />
        </Grid>
      </Grid>
      <br />
      <FormLabel>Paramètres liés aux champs de saisie</FormLabel>
      <br />
      {renderParametersList()}
    </GridItem>
  );
};

UrlBlock.propTypes = {
  engine: shape({}),
  handleChange: func,
};

UrlBlock.defaultProps = {
  engine: null,
  handleChange: Function.Prototype,
};

export default UrlBlock;
