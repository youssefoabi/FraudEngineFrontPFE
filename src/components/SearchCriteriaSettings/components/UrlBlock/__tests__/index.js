import { itRendersAllMutations } from '@cdiscount/jest-utils';

import UrlBlock from '../';

jest.resetModules();

jest.mock('../useUrl', () => ({
  __esModule: true,
  default: jest
    .fn()
    .mockReturnValueOnce({})
    .mockReturnValueOnce({
      allParameters: {},
    })
    .mockReturnValueOnce({
      allParameters: [
        {
          key: 'search',
          typeId: 0,
        },
        {
          key: 'Latitude',
          typeId: 1,
        },
      ],
    }),
  namedExport: 'useUrl',
}));

const mutations = [
  { name: 'with null parameters' },
  { name: 'with empty parameters' },
  { name: 'with parameters' },
];

describe(UrlBlock.name, () => {
  itRendersAllMutations(UrlBlock, mutations);
});
