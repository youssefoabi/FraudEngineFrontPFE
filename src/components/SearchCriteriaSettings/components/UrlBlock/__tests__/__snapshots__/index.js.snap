// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlBlock UrlBlock renders correctly with empty parameters 1`] = `
<h
  fullheight={false}
  title="Gamme "
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      columns={6}
      item={true}
      lg={6}
      md={6}
      xs={6}
    >
      <WithStyles(ForwardRef(FormLabel))>
        Url Pattern
      </WithStyles(ForwardRef(FormLabel))>
      <ScopeUrlField
        scopeUrl={null}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      columns={6}
      item={true}
      lg={6}
      md={6}
      xs={6}
    >
      <f
        config={null}
        fields={
          Array [
            Object {
              "columns": 8,
              "id": "name",
              "label": "Libelle du Radio-Bouton",
              "type": "input",
            },
            Object {
              "columns": 4,
              "defaultValue": false,
              "id": "isEnable",
              "label": "Gamme activée",
              "type": "switch",
            },
          ]
        }
        inputs={null}
        onChange={[Function]}
        onInit={[Function]}
        title={null}
        values={null}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <br />
  <WithStyles(ForwardRef(FormLabel))>
    Paramètres liés aux champs de saisie
  </WithStyles(ForwardRef(FormLabel))>
  <br />
</h>
`;

exports[`UrlBlock UrlBlock renders correctly with null parameters 1`] = `
<h
  fullheight={false}
  title="Gamme "
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      columns={6}
      item={true}
      lg={6}
      md={6}
      xs={6}
    >
      <WithStyles(ForwardRef(FormLabel))>
        Url Pattern
      </WithStyles(ForwardRef(FormLabel))>
      <ScopeUrlField
        scopeUrl={null}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      columns={6}
      item={true}
      lg={6}
      md={6}
      xs={6}
    >
      <f
        config={null}
        fields={
          Array [
            Object {
              "columns": 8,
              "id": "name",
              "label": "Libelle du Radio-Bouton",
              "type": "input",
            },
            Object {
              "columns": 4,
              "defaultValue": false,
              "id": "isEnable",
              "label": "Gamme activée",
              "type": "switch",
            },
          ]
        }
        inputs={null}
        onChange={[Function]}
        onInit={[Function]}
        title={null}
        values={null}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <br />
  <WithStyles(ForwardRef(FormLabel))>
    Paramètres liés aux champs de saisie
  </WithStyles(ForwardRef(FormLabel))>
  <br />
</h>
`;

exports[`UrlBlock UrlBlock renders correctly with parameters 1`] = `
<h
  fullheight={false}
  title="Gamme "
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      columns={6}
      item={true}
      lg={6}
      md={6}
      xs={6}
    >
      <WithStyles(ForwardRef(FormLabel))>
        Url Pattern
      </WithStyles(ForwardRef(FormLabel))>
      <ScopeUrlField
        scopeUrl={null}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      columns={6}
      item={true}
      lg={6}
      md={6}
      xs={6}
    >
      <f
        config={null}
        fields={
          Array [
            Object {
              "columns": 8,
              "id": "name",
              "label": "Libelle du Radio-Bouton",
              "type": "input",
            },
            Object {
              "columns": 4,
              "defaultValue": false,
              "id": "isEnable",
              "label": "Gamme activée",
              "type": "switch",
            },
          ]
        }
        inputs={null}
        onChange={[Function]}
        onInit={[Function]}
        title={null}
        values={null}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <br />
  <WithStyles(ForwardRef(FormLabel))>
    Paramètres liés aux champs de saisie
  </WithStyles(ForwardRef(FormLabel))>
  <br />
  <ParameterChip
    engine={null}
    parameter={
      Object {
        "key": "search",
        "typeId": 0,
      }
    }
    scopeId={null}
  />
  <ParameterChip
    engine={null}
    parameter={
      Object {
        "key": "Latitude",
        "typeId": 1,
      }
    }
    scopeId={null}
  />
</h>
`;
